libdnf5/rpm/solv/goal_private.cpp

```diff
@@ -580,11 +580,8 @@ libdnf5::GoalProblem GoalPrivate::protected_in_removals() {
     auto & pool = get_rpm_pool();
 
     libdnf5::solv::SolvMap pkg_remove_list(pool->nsolvables);
-    for (auto index = 0; index < removes.size(); ++index) {
-        pkg_remove_list.add_unsafe(removes[index]);
-    }
-    for (auto index = 0; index < obsoleted.size(); ++index) {
-        pkg_remove_list.add_unsafe(obsoleted[index]);
+    for (const auto & remove : removes) {
```

Comment:
May I ask you to add a comment that explains why ignoring obsoleted packages is intended.

Reply:
The rest looks good to me.

Reply:
Good call, I added a comment.